### ISSUE TITLE
Is it possible to only return dom elements from the ref callback

### ISSUE THREAD

**Issue by Letladi** [facebook/react#13680]
Hi guys, I'm wondering if it's possible to only return dom elements from ref callbacks. 
In a codebase we are working on currently, we have the need to use to refs in some parts. We find that we have to wrap our components id divs just so we can get a dom element in the ref; this is required by another library we are using (openlayers). 

It would be more convenient if only dom elements were returned in the callback (unless ofcourse there is a reason for having custom component object proxies in the callback; in which case can't we have the dom element attached to the proxy?).

If it's a trivial change, I would be willing (given some guidance of course 🙂) to work on it.

**Comment by gaearon** [comment]
By proxy do you mean an inner class component instance? The suggested solution is for the inner component is to [forward a ref to the DOM node if desired](https://reactjs.org/docs/forwarding-refs.html#forwarding-refs-to-dom-components). Then you won't have a wrapper in the middle.

**Comment by Letladi** [comment]
Something like this;

```js
<div 
    ref={element => {
      // the element in this case is a dom element
}} />
<Vehicle
    ref={element => {
    // I want element here to be the dom element as well, instead of the current behavior where it returns an object
}}
/>
```

**Comment by gaearon** [comment]
Yes, for this to work you'd need to use `forwardRef` in `Vehicle`, like described in the link.

**Comment by Letladi** [comment]
Thanks. I'll give that a go.
